### PR TITLE
demo: Tag demo update

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 无需纳入 ChangeLog |
| 🇨🇳 Chinese | 无需纳入 ChangeLog |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/table/demo/jsx.tsx [hunk 1]
```diff
@@ -53,11 +53,17 @@ const App: React.FC = () => (
...
-          {tags.map((tag) => (
-            <Tag color="blue" key={tag}>
-              {tag}
-            </Tag>
-          ))}
+          {tags.map((tag) => {
+            let color = tag.length > 5 ? 'geekblue' : 'green';
+            if (tag === 'loser') {
+              color = 'volcano';
+            }
```
Comment (MadCcc) [inline components/table/demo/jsx.tsx:60]: 这没啥必要吧

File: components/select/demo/custom-tag-render.tsx [hunk 2]
```diff
@@ -4,7 +4,12 @@ import type { SelectProps } from 'antd';
 
 type TagRender = SelectProps['tagRender'];
 
-const options = [{ value: 'gold' }, { value: 'lime' }, { value: 'green' }, { value: 'cyan' }];
+const options: SelectProps['options'] = [
+  { value: 'gold', label: <span>gold</span> },
```
Comment (MadCcc) [inline components/select/demo/custom-tag-render.tsx:8]: label 一样的就不用写了

File: components/table/demo/jsx.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/table/demo/jsx.tsx:60]: 这里是为了跟这两个地方保持统一：

https://github.com/ant-design/ant-design/blob/3d7a3813edb1ce80425f1634d469fde3288bea3a/components/table/demo/pagination.tsx#L54-L64

https://github.com/ant-design/ant-design/blob/3d7a3813edb1ce80425f1634d469fde3288bea3a/components/table/demo/basic.tsx#L36-L46

File: components/select/demo/custom-tag-render.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/select/demo/custom-tag-render.tsx:8]: 这里指定了 `SelectProps['options']` 类型，必须写 `label`，否则会报错

File: components/table/demo/jsx.tsx (hunk 1 above)
Comment (MadCcc) [reply components/table/demo/jsx.tsx:60]: 一样的那没事了

File: components/select/demo/custom-tag-render.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/select/demo/custom-tag-render.tsx]: revert 了

**Review by afc163 — APPROVED** [review]

**Pull request merged**